x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizerTests.java

```diff
@@ -9084,4 +9136,77 @@ public void testDecayScaleMustBeLiteral() {
+     *     \_EsRelation[languages_lookup][LOOKUP][language_code{f}#16, language_name{f}#17]
+     *
+     */
+    public void testLookupJoinExpressionSwapped() {
+        LogicalPlan plan = optimizedPlan("""
+            from test
+            | keep languages
+            | lookup join languages_lookup ON language_code == languages
+            """);
+        var project = as(plan, Project.class);
+        var limit = asLimit(project.child(), 1000, true);
+        var join = as(limit.child(), Join.class);
+        assertEquals("language_code == languages", join.config().joinOnConditions().toString());
+        var equals = as(join.config().joinOnConditions(), Equals.class);
+        // we expect left and right to be swapped
+        var left = as(equals.left(), Attribute.class);
+        var right = as(equals.right(), Attribute.class);
+        assertEquals("language_code", right.name());
+        assertEquals("languages", left.name());
+        var limitPastJoin = asLimit(join.left(), 1000, false);
+        as(limitPastJoin.child(), EsRelation.class);
+        as(join.right(), EsRelation.class);
+    }
+
+    public void testLookupJoinExpressionAmbigiousRight() {
+        String query = """
+            from test
+            | rename languages as language_code
+            | lookup join languages_lookup ON salary == language_code
+            """;
+        IllegalStateException e = expectThrows(IllegalStateException.class, () -> plan(query));
+        assertThat(e.getMessage(), containsString("Reference [language_code] is ambiguous; matches any of "));
+    }
+
+    public void testLookupJoinExpressionAmbigiousLeft() {
+        String query = """
+            from test
+            | rename languages as language_name
+            | lookup join languages_lookup ON language_name == language_code
+            """;
+        IllegalStateException e = expectThrows(IllegalStateException.class, () -> plan(query));
+        assertThat(e.getMessage(), containsString("Reference [language_name] is ambiguous; matches any of "));
+    }
+
+    public void testLookupJoinExpressionAmbigiousBoth() {
+        String query = """
+            from test
+            | rename languages as language_code
+            | lookup join languages_lookup ON language_code != language_code
+            """;
+        IllegalStateException e = expectThrows(IllegalStateException.class, () -> plan(query));
+        assertThat(e.getMessage(), containsString("Reference [language_code] is ambiguous; matches any of "));
+    }
+
+    public void testLookupJoinExpressionAnd() {
+        String query = """
+            from test
+            | rename languages as language_code_left
+            | lookup join languages_lookup ON language_code_left != language_code and language_code_left > language_code
+            """;
+
+        LogicalPlan plan = optimizedPlan(query);
+        var project = as(plan, Project.class);
```

Comment:
We don't assert a whole lot here. Let's either make the test more specific or axe it?